modules/charon-core/src/main/java/org/wso2/charon3/core/schema/SCIMAttributeSchema.java

```diff
@@ -77,6 +82,7 @@ private SCIMAttributeSchema(String uri, String name, SCIMDefinitions.DataType ty
         this.subAttributes = subAttributes;
         this.canonicalValues = canonicalValues;
         this.referenceTypes = referenceTypes;
+        existingSchemas.put(uri, this);
```

Comment:
'existingSchemas' map should be thread-safe here since SCIMAttributeSchema constructor can be called through createSCIMAttributeSchema method by multiple threads.